app/models/spree/shipping_method.rb

```diff
@@ -13,11 +13,11 @@ class ShippingMethod < ApplicationRecord
 
     default_scope -> { where(deleted_at: nil) }
 
-    has_many :shipping_rates, inverse_of: :shipping_method
+    has_many :shipping_rates, inverse_of: :shipping_method, dependent: :destroy
     has_many :shipments, through: :shipping_rates
-    has_many :shipping_method_categories
-    has_many :shipping_categories, through: :shipping_method_categories
-    has_many :distributor_shipping_methods
+    has_many :shipping_method_categories, dependent: :destroy
+    has_many :shipping_categories, through: :shipping_method_categories, dependent: :destroy
```

Comment:
No, we don't want to delete categories when deleting shipping methods.

Reply:
My understanding is that this code will destroy `shipping_method_categories` (ie. the record that links the two) only and not the actual record in `shipping_categories` when using a `through:` relationship like this, [according to the API documentation](https://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#:~:text=have%20been%20deleted.-,What%20gets%20deleted%3F,-There%20is%20a).

Reply:
Well, you have associations to both models here. You can destroy shipping_method_categories and not destroy shipping_categories. That's what I meant.